temporal/fx.go

```diff
@@ -584,6 +585,7 @@ func ApplyClusterMetadataConfigProvider(
 	logger = log.With(logger, tag.ComponentMetadataInitializer)
 	metricsHandler = metricsHandler.WithTags(metrics.ServiceNameTag(primitives.ServerService))
 	clusterName := persistenceClient.ClusterName(svc.ClusterMetadata.CurrentClusterName)
+	serializer := serialization.NewSerializer()
```

Comment:
do we need to get this serializer from fx graph?

Reply:
Maybe I'm a bit confused about the fx graph, I find it hard to follow. I've made this change, not sure it addresses your concern. https://github.com/temporalio/temporal/pull/8222/commits/8acd951cb5f2528f81aee7d09f2c97580bb45108

Reply:
Also noticed one more; addressed in https://github.com/temporalio/temporal/pull/8222/commits/045e533c2c6e2db75e3f05843e5de529d3dd7512

Reply:
oh if we do that we should just make serializer a top level module, otherwise it's very confusing as it's provided again later in the common/resource module.

I am fine either way (revert the 045e533 and just create one inline, sorry!) or make it top level.